turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -2319,16 +2336,18 @@ async fn handle_free_var_reference(
         ),
 
         FreeVarReference::Value(value) => {
-            analysis.add_code_gen(ConstantValue::new(
-                Value::new(value.clone()),
-                Vc::cell(ast_path.to_vec()),
-            ));
+            analysis.add_code_gen(
+                ConstantValue::new(Value::new(value.clone()), Vc::cell(ast_path.to_vec()))
+                    .to_resolved()
+                    .await?,
```

Comment:
For concurrency it would probably be better to collect `Vcs` and call `to_resolved()` after the analysis is finished on the whole lists of Vcs.

Reply:
This isn't actually the code generation happening here, that happens later when `CodeGeneratable::code_generation` is called.

That said, `AnalyzeEcmascriptModuleResultBuilder` isn't a `#[turbo_tasks::value]`, so as you say in your later comment I can resolve this in `build()`.